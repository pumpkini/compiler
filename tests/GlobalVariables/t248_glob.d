int [] distance;
int m;
int [] dest;
int [] src;
int [] weight;
	
void belman_ford() {
	int i;
	for(i = 0;i < m;i = i + 1) {
		relax(i);
	}
}

void relax(int index) {
	if (distance[src[index]] + weight[index] < distance[dest[index]]) {
		distance[dest[index]] = distance[src[index]] + weight[index];
	}
}

int main() {
	int n;
	int i;
	n = ReadInteger();
	m = ReadInteger();
	distance = NewArray(n, int);
	dest = NewArray(m, int);
	src = NewArray(m, int);
	weight = NewArray(m, int);
	for(i = 0;i < m;i = i + 1) {
		Print("Write ", i, "th edge:");
		src[i] = ReadInteger();
		dest[i] = ReadInteger();
		weight[i] = ReadInteger();
	}
	for(i = 0;i < n;i = i + 1) {
		if (i == 0) distance[i] = 0;
		else distance[i] = 1000000;
	}
	for(i = 0;i < n;i = i + 1) {
		belman_ford();
	}
	for(i = 0;i < n; i = i + 1) 
		Print("Distance of ", i, "th node: ", distance[i]);
}