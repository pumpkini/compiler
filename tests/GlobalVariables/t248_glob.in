3
3
0
1
3
1
2
6
2
0
6