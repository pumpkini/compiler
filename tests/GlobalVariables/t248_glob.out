Write 0th edge:
Write 1th edge:
Write 2th edge:
Distance of 0th node: 0
Distance of 1th node: 3
Distance of 2th node: 9
